csharp/src/Drivers/Apache/ApacheUtility.cs

```diff
@@ -0,0 +1,77 @@
+﻿/*
+* Licensed to the Apache Software Foundation (ASF) under one or more
+* contributor license agreements.  See the NOTICE file distributed with
+* this work for additional information regarding copyright ownership.
+* The ASF licenses this file to You under the Apache License, Version 2.0
+* (the "License"); you may not use this file except in compliance with
+* the License.  You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using System.Threading;
+
+namespace Apache.Arrow.Adbc.Drivers.Apache
+{
+    internal class ApacheUtility
+    {
+        internal const int QueryTimeoutSecondsDefault = 60;
+
+        public enum TimeUnit
+        {
+            Seconds,
+            Milliseconds
+        }
+
+        public static CancellationToken GetCancellationToken(int timeout, TimeUnit timeUnit)
+        {
+            TimeSpan span;
+
+            if (timeout == 0 || timeout == int.MaxValue)
+            {
+                // the max TimeSpan for CancellationTokenSource is int.MaxValue in milliseconds (not TimeSpan.MaxValue)
+                // no matter what the unit is
+                span = TimeSpan.FromMilliseconds(int.MaxValue);
```

Comment:
@davidhcoe There is a dedicated field/constant that represents infinite in TimeSpan - do we want to use this, instead?
https://learn.microsoft.com/en-us/dotnet/api/system.threading.timeout.infinitetimespan?view=netframework-4.7.2